server/datastore/mysql/hosts.go

```diff
@@ -981,15 +981,24 @@ func (ds *Datastore) TotalAndUnseenHostsSince(ctx context.Context, daysCount int
 		Total  int `db:"total"`
 		Unseen int `db:"unseen"`
 	}
+
+	// convert daysCount to integer number of seconds for more precision in sql query
+	dur, err := time.ParseDuration(fmt.Sprintf("%dh", daysCount*24))
+	if err != nil {
+		return 0, 0, ctxerr.Wrap(ctx, err, "getting total and unseen host counts")
+	}
+	unseenInterval := int(dur.Seconds())
+
 	err = sqlx.GetContext(ctx, ds.reader, &counts,
 		`SELECT
 			COUNT(*) as total,
-			SUM(IF(DATEDIFF(CURRENT_DATE, COALESCE(hst.seen_time, h.created_at)) >= ?, 1, 0)) as unseen
+			SUM(IF(TIMESTAMPDIFF(SECOND, COALESCE(hst.seen_time, h.created_at), CURRENT_TIMESTAMP) >= ?, 1, 0)) as unseen
```

Comment:
Instead of `CURRENT_TIMESTAMP`, I think this should be `UNIX_TIMESTAMP`, since the latter is seconds since epoch, and the former is `YYYYMMDDhhmmss` as a number.

See https://www.w3resource.com/mysql/date-and-time-functions/mysql-current_timestamp-function.php vs https://www.w3resource.com/mysql/date-and-time-functions/mysql-unix_timestamp-function.php

Reply:
I believe that the `TIMESTAMPDIFF` function expects the `YYYYMMDDhhmmss` format. Switching the code to `UNIX_TIMESTAMP` causes the tests to fail.

Reply:
Hm, I think you're right. Kind of hard to reason about.